system-x/services/telegram/src/main/java/software/tnb/telegram/validation/TelegramValidation.java

```diff
@@ -20,25 +24,31 @@ public TelegramValidation(Telegram client) {
     }
 
     public void sendMessage(String text) {
-        LOG.debug("Send message " + text + " from telegram-client ");
+        LOG.debug("Sending message {} from telegram-client", text);
         try {
             client.execInContainer("python3", "/app/send_message.py", text);
         } catch (Exception e) {
-            fail("Failed to send message", e);
+            throw new RuntimeException("Failed to send message", e);
         }
     }
 
-    public List<String> getLastNMessages(int n) {
+    public List<Message> getLastNMessages(int n) {
         LOG.debug("Get last " + n + " messages: ");
-        List<String> messages;
         try {
-            String messagesInStr = client.execInContainer("python3", "/app/get_messages.py", n + "");
-            LOG.debug(messagesInStr);
-            messages = Arrays.asList(messagesInStr.split("\n"));
+            String response = client.execInContainer("python3", "/app/get_messages.py", n + "");
+            LOG.debug(response);
```

Comment:
how about `LOG.debug("Received messages: {}", response);` so there's some context for the JSON?